include/boost/beast/experimental/core/uri/parts.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2018 oneiric (oneiric at i2pmail dot org)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_URI_PARTS_HPP
+#define BOOST_BEAST_URI_PARTS_HPP
+
+#include <boost/beast/core/string.hpp>
+
+namespace boost {
+namespace beast {
+namespace uri   {
+
+class parts {
+  std::string scheme_;
+  std::string username_;
+  std::string password_;
+  std::string host_;
+  std::string port_;
+  std::string path_;
+  std::string query_;
+  std::string fragment_;
+
+protected:
+  void reset()
+  {
+    scheme_.clear();
+    username_.clear();
+    password_.clear();
+    host_.clear();
+    port_.clear();
+    path_.clear();
+    query_.clear();
+    fragment_.clear();
+  }
+
+public:
+  void scheme(std::string part) { scheme_ = part; }
+  std::string scheme() const noexcept { return scheme_; }
```

Comment:
Why use a std::string? Can be use a string_view(C++17/boost.beast).

Reply:
I chose to use std::string because use in Kovri was a bit simpler that way. After a little bit of work, the performance benefits of string_view are worth converting to string when necessary.